### DISCUSSION TITLE
Refresh the page and a relink popup will appear

### DISCUSSION THREAD

**Discussion by lrlazy** [WalletConnect/walletconnect-monorepo#3462] [Developer Support]
When I refresh the page, the relink popup appears. But my wallet is still in the link state, and I will fail to re-link

**Comment by glitch-txs** [comment]
Hi! What do you mean by "re-link"?

**Reply by lrlazy** [reply]
Scan the QR code again

**Reply by lrlazy** [reply]
I use the following code
await EthereumProvider.init({
   projectId: process.env.TEST_PROJECT_ID,
   chains: [1], // chains added to required namespaces
   optionalChains: [5] // chains added to optional namespaces
   ...
})

How should I be able to auto-link my already linked after refreshing the page

**Reply by glitch-txs** [reply]
just call this function again, the provider will now have a `session` property, you can also just request the account method right away. e.g.

```ts
if(provider.session){
const accounts = provider.request({ method: 'eth_accounts' })
}
```

**Reply by lrlazy** [reply]
It has been automatically linked according to the above method, but the following error message will be reported when requesting 'eth_sendTransaction'
Socket stalled when trying to connect to wss://relay.walletconnect.com

**Reply by glitch-txs** [reply]
could you share a minimal reproduction of the issue?

**Reply by lrlazy** [reply]
how should i share with you

**Reply by glitch-txs** [reply]
A small repo or something like StackBlitz should do
